### ISSUE TITLE
Average error and PCK

### ISSUE THREAD

**Issue by ZhangMaom** [bardiadoosti/HOPE#17]
Hi, I cloned the code and ran the pretrained model.
`python HOPE.py   --input_file ./datasets/fhad/   --test   --batch_size 32   --model_def HopeNet   --gpu   --gpu_number 0   --pretrained_model ./checkpoints/fhad/model-0.pkl`
The test loss is 253.
Then I calculated the average error and PCK of the pretrained model.
Average error is `33.6769 mm`.
For PCK, I got `10mm@ 14.78%, 20mm@ 42.79%, 30mm@ 64.81%, 40mm@ 77.59%, 50mm@ 84.35%`
(torch==1.1.0, torchvision==0.3.0)
It seems different from your results in the paper.
Mabe I got something wrong. Can you provide the average error and PCK of the pretrained model.
Hope for your reply. Thank you.

**Comment by bardiadoosti** [comment]
Hi,

Thanks for your comment. The result shown in the paper are from ResNet10 which worked better on FPHA dataset in case of overfitting for initial 2D estimation. But released model belongs to ResNet50 which in general has better performance than ResNet10 especially on bigger datasets. But the 3D performance which was the main contribution of this paper were similar.

**Comment by ZhangMaom** [comment]
Thanks for your reply. 
In fact, the question I want to ask is about 3D performance. But I gave 2D performance above, which is my fault. So I reedited it.
The 3D performance of the pretrained model I got is as follows.
The average error on 3D hand pose estimation given RGB image is 18.8588mm.
For the percentage of correct 3D hand pose of HopeNet on fhad, I got `10mm@ 28.71%, 20mm@ 69.02%, 30mm@ 85.82%, 40mm@92.32%, 50mm@ 95.62%`.
I can not compare the average error of the HopeNet as there is only average error of AdaptiveGraphUNet in the paper. 
However, for PCK, there seems to be a big difference between the results in the paper and the results of the pretrained model I got. Using ResNet50 instead of ResNet10 will have some effect on the results, but the effect seems to be somewhat large.

In addition, I ran the GraphUNet given 2D pose with the parameters in the pretrained model. 
The test loss I got is 1966. The 3D performance was even worse. I don't know why.
Hope for your reply. Thank you.

**Comment by mslqing** [comment]
@ZhangMaom Hello, did you write your PCK program by yourself? I can't find a PCK program in the author's code. If you wrote it yourself, can you share it?

**Comment by bardiadoosti** [comment]
> Thanks for your reply.
> In fact, the question I want to ask is about 3D performance. But I gave 2D performance above, which is my fault. So I reedited it.
> The 3D performance of the pretrained model I got is as follows.
> The average error on 3D hand pose estimation given RGB image is 18.8588mm.
> For the percentage of correct 3D hand pose of HopeNet on fhad, I got `10mm@ 28.71%, 20mm@ 69.02%, 30mm@ 85.82%, 40mm@92.32%, 50mm@ 95.62%`.
> I can not compare the average error of the HopeNet as there is only average error of AdaptiveGraphUNet in the paper.
> However, for PCK, there seems to be a big difference between the results in the paper and the results of the pretrained model I got. Using ResNet50 instead of ResNet10 will have some effect on the results, but the effect seems to be somewhat large.
> 
> In addition, I ran the GraphUNet given 2D pose with the parameters in the pretrained model.
> The test loss I got is 1966. The 3D performance was even worse. I don't know why.
> Hope for your reply. Thank you.

I will try to prepare a docker file with the exact condition (anaconda, pytorch and other libraries versions) that I tested my code.

**Comment by bardiadoosti** [comment]
> @ZhangMaom Hello, did you write your PCK program by yourself? I can't find a PCK program in the author's code. If you wrote it yourself, can you share it?

The PCK diagram code that we wrote is similar to Hasson et al.'s Obman paper code:
https://github.com/hassony2/obman_train/blob/master/mano_train/evaluation/zimeval.py

**Comment by ZhangMaom** [comment]
> @ZhangMaom Hello, did you write your PCK program by yourself? I can't find a PCK program in the author's code. If you wrote it yourself, can you share it?

My code is also similar to Hasson et al.'s Obman paper code:
[https://github.com/hassony2/obman_train/blob/master/mano_train/evaluation/zimeval.py](url)

**Comment by ZhangMaom** [comment]
> I will try to prepare a docker file with the exact condition (anaconda, pytorch and other libraries versions) that I tested my code.

Thanks! Looking forward to it!

**Comment by mslqing** [comment]
Hello, I used your model to test, and the results of the pck curve obtained are different from yours. The threshold is 10mm: 24.79%, 20mm: 62.55%, 30mm: 81.22%, 40mm: 89.56%, 50mm : 93.7%.
Can you share your obman's pre-trained model?

**Comment by ZhangMaom** [comment]
@mslqing Did you try to reproduce the results in the paper

**Comment by mslqing** [comment]
@ZhangMaom  I reproduce the code results in the paper as 10mm: 24.79%, 20mm: 62.55%, 30mm: 81.22%, 40mm: 89.56%, 50mm: 93.7%. It is very different from the results in the paper. Can you get the results in the paper?

**Comment by ZhangMaom** [comment]
> @ZhangMaom  I reproduce the code results in the paper as 10mm: 24.79%, 20mm: 62.55%, 30mm: 81.22%, 40mm: 89.56%, 50mm: 93.7%. It is very different from the results in the paper. Can you get the results in the paper?

The result I reproduced is also different from the results in the paper.

**Comment by mslqing** [comment]
@ZhangMaom Hello,  I saw your results in the discussion above. I am Chinese. Give me your email address. Let's discuss it together.My email: msl_qing@163.com

**Comment by mslqing** [comment]
@ZhangMaom Hello, I saw your results in the discussion above. I am Chinese. Can you give me your email address? Let's discuss it together.My email: msl_qing@163.com

**Comment by bardiadoosti** [comment]
> Hi,
> 
> Thanks for your comment. The result shown in the paper are from ResNet10 which worked better on FPHA dataset in case of overfitting for initial 2D estimation. But released model belongs to ResNet50 which in general has better performance than ResNet10 especially on bigger datasets. But the 3D performance which was the main contribution of this paper were similar.

I think I have to quote my comment again. It is clearly mentioned in the paper that the results belong to ResNet10 and this weights are for ResNet50.

**Comment by bardiadoosti** [comment]
Hi, today I checked the weights and found out unfortunately something is wrong with the provided weights specially regarding to the GraphUNet's performance. I try to solve this problem by providing the right weights for the GraphUNet in the next couple of days. I apologize for any inconvenience.

**Comment by 3tuuu** [comment]
> 

hello, I'm a newbie, and dont know how to calculated PCK or PCP of the pretrained model  ?  can you give me some code example?
